models/qrCodeAuth.js

```diff
@@ -41,6 +41,13 @@ const storeUserDeviceInfo = async (userDeviceInfoData) => {
   try {
     const { user_id: userId } = userDeviceInfoData;
     const user = await userModel.doc(userId).get();
+
+    if (!user.data()) {
+      return {
+        userExists: false,
+      };
+    }
```

Comment:
please use `dataAccessLayer` to fetch the user details. Since the user model contains sensitive data.

Reply:
Did not get this point, can u elaborate more on this?

Reply:
added dataAccessLayer

Reply:
addressed